src/SecureStore.CyberArkCCP/CyberArkCCPSecureStore.cs

```diff
@@ -0,0 +1,267 @@
+                    {
+                        return ctx;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    normalizationException = ex;
+                }
+            }
+
+            throw new SecureStoreException(
+                    SecureStoreException.Type.InvalidConfiguration,
+                    SecureStoresUtil.GetLocalizedResource(nameof(Resource.SecureStore)),
+                    normalizationException,
+                    ErrorCode.InvalidSecureStoreContext);
+        }
+
+        private Task<CyberArkCCPPassword> ReadFromCyberArkCCP(string context, string key)
+        {
+            SecureStoresUtil.ThrowIfNull(key);
+            var ctx = ThrowIfInvalidContext(context);
+
+            X509Certificate2Collection certCollection = new X509Certificate2Collection();
+
+            if (_httpClientHandler.ClientCertificates.Count == 0)
+            {
+                certCollection = GetCertificateForCCPAuth(ctx.Thumbprint);
+                Exception normalizationException = null;
+
+                if (certCollection.Count != 1)
+                {
+                    throw new SecureStoreException(
+                        SecureStoreException.Type.InvalidConfiguration,
+                        SecureStoresUtil.GetLocalizedResource(nameof(Resource.SecureStoreCert)),
+                        normalizationException,
+                        ErrorCode.InvalidSecureStoreContext);
+                }
+                if (!certCollection[0].HasPrivateKey)
+                {
+                    throw new SecureStoreException(
+                        SecureStoreException.Type.InvalidConfiguration,
+                        SecureStoresUtil.GetLocalizedResource(nameof(Resource.SecureStoreCert)),
+                        normalizationException,
+                        ErrorCode.InvalidSecureStoreContext);
+                }
+                _httpClientHandler.ClientCertificates.Add(certCollection[0]);
+            }
+
+            return GetPasswordFromCCP(ctx, key);
+        }
+
+        private async Task<CyberArkCCPPassword> GetPasswordFromCCP(CyberArkCCPOptions ctx, string key)
+        {
+            Exception normalizationException = null;
+            try
+            {
+                string obj;
+                key = key.Replace('\\', '-');
+                if (string.IsNullOrEmpty(ctx.Folder))
+                {
+                    obj = $"{key}";
+                }
+                else
+                {
+                    obj = $"{ctx.Folder}\\{key}";
+                }
+
+                var uri = new Uri(ctx.URL + $"/AIMWebService/api/Accounts?appid={ctx.ApplicationId}&safe={ctx.Safe}&object={obj}");
+                var response = await _httpClient.GetAsync(uri);
+                var content = await response.Content.ReadAsStringAsync();
+
+                if (content.Contains("ErrorCode"))
```

Comment:
Let's first deserialize it as `CyberArkCCPPassword` and if the `Content` property is null or empty then we should try deserialize as as an `CyberArkCCPError`